include/rx/ranges.hpp

```diff
@@ -1807,6 +1807,33 @@ struct none_of {
 };
 template <class P>
 none_of(P &&)->none_of<P>;
+    
+template <class P>
+struct contains {
```

Comment:
It would be nice to add a customization point for the comparison function, defaulting to `std::equal_to<void>`, and using "empty base class optimization". Take a look at `uniq()` for how this can be achieved.

Reply:
I'll take a look

Reply:
Wouldn't custom comparison function just make it the second any_of? That's unnecessary complexity.

Reply:
Well, it would be a version of `any_of` that doesn't require the user to pass a lambda function that captures the value.